discovery-infra/utils.py

```diff
@@ -230,11 +250,16 @@ def file_exists(file_path):
     return Path(file_path).exists()
 
 
-def recreate_folder(folder):
+def recreate_folder(folder, on_create=None):
     if os.path.exists(folder):
         shutil.rmtree(folder)
+
     os.makedirs(folder, exist_ok=True)
-    run_command("chmod ugo+rx %s" % folder)
+    run_command("chmod -R ugo+rx %s" % folder)
+
+    if on_create is None:
```

Comment:
I find this feature a little bit redundant. It can run after this function has finished and the folder has created.